src/aatoolbox/datasources/ipc/ipc.py

```diff
@@ -0,0 +1,170 @@
+    "#.8": "ML1_1",
+    "%.6": "ML1_1_perc",
+    "#.9": "ML1_2",
+    "%.7": "ML1_2_perc",
+    "#.10": "ML1_3",
+    "%.8": "ML1_3_perc",
+    "#.11": "ML1_4",
+    "%.9": "ML1_4_perc",
+    "#.12": "ML1_5",
+    "%.10": "ML1_5_perc",
+    "#.13": "ML1_3p",
+    "%.11": "ML1_3p_perc",
+    "#.14": "reported_pop_ML2",
+    "Area Phase.2": "ML2_phase",
+    "Analysis Period.2": "ML2_val",
+    "#.15": "ML2_1",
+    "%.12": "ML2_1_perc",
+    "#.16": "ML2_2",
+    "%.13": "ML2_2_perc",
+    "#.17": "ML2_3",
+    "%.14": "ML2_3_perc",
+    "#.18": "ML2_4",
+    "%.15": "ML2_4_perc",
+    "#.19": "ML2_5",
+    "%.16": "ML2_5_perc",
+    "#.20": "ML2_3p",
+    "%.17": "ML2_3p_perc",
+}
+
+
+def _preprocess_raw_data(
+    raw_file_path: Path,
+    output_path: Path,
+):
+    """
+    Preprocess the downloaded data by checking.
+
+    if the file contains data and changing column names
+    :param raw_file_path: path to the file with the data to read
+    :param output_path: path to write the preprocessed file to
+    """
+    df = pd.read_excel(
+        raw_file_path,
+        # on previous rows there is logo etc
+        header=[11],
+    )
+
+    # only select rows with data, i.e. have a date of analysis
+    # often have some rows with unneeded info for analysis
+    # e.g. the disclaimer
+    df = df[df["Date of Analysis"].notnull()]
+    if df.empty:
+        # a file from the url is downloaded regardless of whether
+        # the iso2 is corrector ipc has done any analyses
+        # thus check if there is actual data
+        # we might want to check for correctness of iso2 already
+        # before downloading (or when setting-up the config)
+        raise RuntimeError(
+            "Downloaded file doesn't contain any data. "
+            "Make sure your iso2 code is correct."
+        )
+
+    # ipc excel file comes with horrible column names, so change them to
+    # better understandable ones
+    df = df.rename(columns=IPC_COLUMN_NAME_MAPPING)
+    # due to excel settings, the percentages are on the 0 to 1 scale so
+    # change to 0-100
+    perc_cols = [c for c in df.columns if "perc" in c]
+    df[perc_cols] = df[perc_cols] * 100
+    df["date"] = pd.to_datetime(df["date"])
+
+    # write preprocessed data to file
+    df.to_csv(output_path, index=False)
+
+
+def download_ipc(iso3: str, iso2: str, output_dir: Union[Path, str]):
+    """
+    Retrieve the IPC data from their Population Tracking Tool.
+
+    Also do some preprocessing and save the outputs
+    Since the data size per country is very small, we always
+    download all data for the given country
+    :param iso3: iso3 code of country of interest
+    :param iso2: iso2 code of country of interest
+    :param output_dir: path to directory the file should be saved to
+
+    Examples
+    --------
+    >>> download_fewsnet(iso3="som",iso2="so,output_dir=output_dir)
+    """
+    # convert to path object if str
+    if isinstance(output_dir, str):
```

Comment:
Hmm not sure if I understand you correctly. Cause later on I am doing `raw_output_path = output_dir / f"{iso3}_ipc_raw.xlsx"` and if `output_dir` is then a `str` instead of a `Path`, it will complain. What is it what you mean?